### ISSUE TITLE
Uncaught TypeError: defaultOption is undefined

### ISSUE THREAD

**Issue by yuvipanda** [yuvipanda/jupyterhub-fancy-profiles#38]
I was trying to make progress on https://github.com/NASA-IMPACT/veda-jupyterhub/issues/34, and deployed this project to the staging NASA-VEDA hub. It didn't render unfortunately, with the following error:

```
Uncaught TypeError: defaultOption is undefined
    ResourceSelect ResourceSelect.jsx:15
    React 9
    workLoop scheduler.development.js:266
    flushWork scheduler.development.js:239
    performWorkUntilDeadline scheduler.development.js:533
    js scheduler.development.js:571
    js scheduler.development.js:633
    Webpack 13
ResourceSelect.jsx:15:37
```

Here's a minial diff to the `jupyterhub_config.py` file that I needed to do to replicate this locally:

```diff
diff --git a/jupyterhub_config.py b/jupyterhub_config.py
index 334d5d2..3fa48f8 100644
--- a/jupyterhub_config.py
+++ b/jupyterhub_config.py
@@ -224,3 +224,25 @@ c.KubeSpawner.profile_list = [
         },
     },
 ]
+
+c.KubeSpawner.profile_list = [
+    {
+        "description": "Specify your own docker image (must have python and jupyterhub installed in it)",
+        "display_name": "Bring your own image",
+        "profile_options": {
+            "image": {
+                "choices": {},
+                "display_name": "Image",
+                "unlisted_choice": {
+                    "display_name": "Custom image",
+                    "enabled": False,
+                    "kubespawner_override": {"image": "{value}"},
+                    "validation_message": "Must be a publicly available docker image, of form <image-name>:<tag>",
+                    "validation_regex": "^.+:.+$",
+                    "display_name_in_choices": "Other...",
+                },
+            },
+        },
+        "slug": "custom",
+    },
+]

+]
```

**Comment by yuvipanda** [comment]
I think it needs to handle the case when `choices` is empty, which is valid when `unlisted_choice.enabled` is True. In such cases, it should not render the select and only render the textbox.

**Comment by batpad** [comment]
Fixed by #41 . I'll close the issue once we've deployed and confirmed it works.
